src/mapreduce.jl

```diff
@@ -160,25 +160,76 @@ end
 ## reducedim ##
 ###############
 
-@inline reducedim(op, a::StaticArray, ::Val{D}) where {D} = mapreducedim(identity, op, a, Val{D})
-@inline reducedim(op, a::StaticArray, ::Val{D}, v0) where {D} = mapreducedim(identity, op, a, Val{D}, v0)
+@inline reducedim(op, a::StaticArray, ::Type{Val{D}}) where {D} = mapreducedim(identity, op, a, Val{D})
```

Comment:
But having to use `Val(D)` in `reducedim` and `Val{D}` in `mapreducedim` seems confusing and inconsistent.  I don't see the use of `Val(D)` and `::Val{D}` anywhere else in StaticArrays, either.  Wouldn't it be better to handle this in a separate PR collectively?